app/streaming/video/ffmpeg-renderers/d3d11va.cpp

```diff
@@ -150,11 +174,130 @@ D3D11VARenderer::~D3D11VARenderer()
+
+        // Magic constants to convert to fixed point.
+        // https://docs.microsoft.com/en-us/windows/win32/api/dxgi1_5/ns-dxgi1_5-dxgi_hdr_metadata_hdr10
+        static constexpr int kMinLuminanceFixedPoint = 10000;
+
+        streamHDRMetaData.RedPrimary[0] = hdrMetadata.displayPrimaries[0].x;
+        streamHDRMetaData.RedPrimary[1] = hdrMetadata.displayPrimaries[0].y;
+        streamHDRMetaData.GreenPrimary[0] = hdrMetadata.displayPrimaries[1].x;
+        streamHDRMetaData.GreenPrimary[1] = hdrMetadata.displayPrimaries[1].y;
+        streamHDRMetaData.BluePrimary[0] = hdrMetadata.displayPrimaries[2].x;
+        streamHDRMetaData.BluePrimary[1] = hdrMetadata.displayPrimaries[2].y;
+        streamHDRMetaData.WhitePoint[0] = hdrMetadata.whitePoint.x;
+        streamHDRMetaData.WhitePoint[1] = hdrMetadata.whitePoint.y;
+        streamHDRMetaData.MaxMasteringLuminance = hdrMetadata.maxDisplayLuminance * kMinLuminanceFixedPoint;
+        streamHDRMetaData.MinMasteringLuminance = hdrMetadata.minDisplayLuminance;
+
+        streamHDRMetaData.MaxContentLightLevel = hdrMetadata.maxContentLightLevel;
+        streamHDRMetaData.MaxFrameAverageLightLevel = hdrMetadata.maxFrameAverageLightLevel;
+        if(streamHDRMetaData.MaxContentLightLevel == 0){
+            streamHDRMetaData.MaxContentLightLevel = streamHDRMetaData.MaxFrameAverageLightLevel;
+        }
+
+        // [TODO] (Source: https://github.com/xbmc) For AMD/HDR,
+        // we apparently need to do a custom tone (MaxMasteringLuminance=10000, MinMasteringLuminance=0).
+        // Yet to be verified
+        // if(m_VideoEnhancement->isVendorAMD()){
+        //     m_StreamHDRMetaData.MaxMasteringLuminance = 10000;
+        //     m_StreamHDRMetaData.MinMasteringLuminance = 0;
+        // }
+
+        // Set HDR Stream (input) Meta data
+        m_VideoContext->VideoProcessorSetStreamHDRMetaData(
+            m_VideoProcessor,
+            0,
+            DXGI_HDR_METADATA_TYPE_HDR10,
+            sizeof(DXGI_HDR_METADATA_HDR10),
+            &streamHDRMetaData
+            );
+    }
+}
+
+/**
+ * \brief Set Monitor HDR MetaData information
+ *
+ * Get the Monitor HDT MetaData via LimeLight library
+ *
+ * \param PSS_HDR_METADATA* HDRMetaData The varaible to set the metadata information
+ * \return bool Return True is succeed
+ */
+void D3D11VARenderer::setHDROutPut(){
+
+    DXGI_HDR_METADATA_HDR10 outputHDRMetaData;
+
+    // Find the monitor attached to the application
+    Microsoft::WRL::ComPtr<IDXGIOutput> pOutput;
+    if (SUCCEEDED(m_SwapChain->GetContainingOutput(&pOutput))) {
+        Microsoft::WRL::ComPtr<IDXGIOutput6> pOutput6;
+        if (SUCCEEDED(pOutput.As(&pOutput6))){
+            DXGI_OUTPUT_DESC1 desc1 {};
+            if (SUCCEEDED(pOutput6->GetDesc1(&desc1))){
+                // Get Monitor ColorSpace for SDR and HDR (if the monitor is capable of HDR)
+                m_OutputColorSpace = desc1.ColorSpace;
```

Comment:
The VideoProcessor's output colorspace needs to match the swapchain's colorspace, not the monitor's. That may be why you're getting incorrect output when HDR enabled state doesn't match between the host and client.

Reply:
I could add shaders to the pipeline to fix color issues. Color spaces are now matching.